org.eclipse.xtext.xtext.ui.examples/projects/arithmetics/org.eclipse.xtext.example.arithmetics.ui.tests/src/org/eclipse/xtext/example/arithmetics/ui/tests/autoedit/AutoEditTest.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2015, 2019 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.example.arithmetics.ui.tests.autoedit;
+
+import java.util.Collections;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.xtext.example.arithmetics.ui.tests.ArithmeticsUiInjectorProvider;
+import org.eclipse.xtext.resource.FileExtensionProvider;
+import org.eclipse.xtext.testing.InjectWith;
+import org.eclipse.xtext.testing.XtextRunner;
+import org.eclipse.xtext.ui.XtextProjectHelper;
+import org.eclipse.xtext.ui.editor.XtextEditor;
+import org.eclipse.xtext.ui.testing.AbstractAutoEditTest;
+import org.eclipse.xtext.ui.testing.util.JavaProjectSetupUtil;
+import org.eclipse.xtext.ui.util.JREContainerProvider;
+import org.eclipse.xtext.ui.util.PluginProjectFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.inject.Inject;
+
+@RunWith(XtextRunner.class)
+@InjectWith(ArithmeticsUiInjectorProvider.class)
+public class AutoEditTest extends AbstractAutoEditTest {
+	@Test
+	public void testCalculatorAutoEdit() throws Exception {
+		String model =
+				"module MyModule\n" +
+				"\n" +
+				"1 + 2;|\n";
+		XtextEditor editor = openEditor(model);
+		pressKey(editor, '\n');
+		String expectation =
+				"module MyModule\n" +
+				"\n" +
+				"1 + 2;\n" +
+				"// = 3\n" +
+				"|\n";
+		assertState(editor, expectation);
+	}
+
+	@Inject
+	private FileExtensionProvider extensionProvider;
```

Comment:
If this wasn't a test I'd remark that the fields should be at the top of the class.